ddl/ddl_api.go

```diff
@@ -2554,7 +2562,7 @@ func modifiableCharsetAndCollation(toCharset, toCollate, origCharset, origCollat
 // change or check existing data in the table.
 // It returns true if the two types has the same Charset and Collation, the same sign, both are
 // integer types or string types, and new Flen and Decimal must be greater than or equal to origin.
-func modifiable(origin *types.FieldType, to *types.FieldType) error {
+func modifiable(origin *types.FieldType, to *types.FieldType, needRewriteCollationData bool) error {
```

Comment:
Ditto, also rewrite the comment of this function.

Reply:
Addressed.